ebpfapi/dllmain.cpp

```diff
@@ -21,8 +21,10 @@ DllMain(HMODULE hModule, unsigned long ul_reason_for_call, void* lpReserved)
         }
         break;
     case DLL_THREAD_ATTACH:
+        ebpf_api_thread_local_initialize();
         break;
     case DLL_THREAD_DETACH:
+        ebpf_api_thread_local_cleanup();
```

Comment:
question: Is `DLL_THREAD_DETACH` guaranteed to be called for all the existing threads when the DLL is been unloaded?

Reply:
This is what the MSDN says:
<html>
<body>
<!--StartFragment-->
DLL_THREAD_DETACH3 | A thread is exiting cleanly. If the DLL has stored a pointer to allocated memory in a TLS slot, it should use this opportunity to free the memory. The system calls the entry-point function of all currently loaded DLLs with this value. The call is made in the context of the exiting thread.
-- | --


<!--EndFragment-->
</body>
</html>